### PR TITLE
add proxy, packageRepository and certificate injection

### LINKED ISSUE #409: Update OnyxiaValues to allow specific injections in region
The api allow three new configuration that the admin could specify at installation.

https://github.com/InseeFrLab/onyxia-api/blob/master/onyxia-model/src/main/java/fr/insee/onyxia/model/region/Region.java#L36

```
    private ProxyInjection proxyInjection;
    @Schema(description = "")
    private PackageRepositoryInjection packageRepositoryInjection;
    @Schema(description = "")
    private CertificateAuthorityInjection certificateAuthorityInjection;
```

This 3 objects should reach the OnyxiaValues https://github.com/InseeFrLab/onyxia-web/blob/2bef7775601f421b9fa4be0512ad4310cc780e0c/src/core/ports/OnyxiaApiClient.ts#L154-L210
in order to be if needed injected in the helm charts if values.schema.json allow it.

As sspcloud don't use this values, the installation is empty for this values, we probably should take care of https://github.com/InseeFrLab/onyxia-web/blob/2bef7775601f421b9fa4be0512ad4310cc780e0c/src/core/secondaryAdapters/mockOnyxiaApiClient.ts?

### PR DESCRIPTION
**Pull request by alexisdondon**
add proxyInjection, packageRepositoriesInjection, CustomAuthorityCertificationInjection that had been added in the onyxia-api by https://github.com/InseeFrLab/onyxia-api/pull/163 in the OnyxiaValues.
That will allow the injection of this values in the helm charts launcher for compatible catalogs.

### CODE REVIEW

File: src/core/secondaryAdapters/officialOnyxiaApiClient.ts [hunk 1]
```diff
@@ -295,6 +308,33 @@ export function createOfficialOnyxiaApiClient(params: {
...
+                        })(),
+                        "certificateAuthorityInjection": (() => {
+                            const { certificateAuthorityInjection } = region ?? {};
+
+                            if (certificateAuthorityInjection === undefined) {
+                                return undefined;
+                            }
+
+                            return { ...certificateAuthorityInjection };
+                        })(),
```
Comment (garronej) [inline src/core/secondaryAdapters/officialOnyxiaApiClient.ts]:
```suggestion
                        "proxyInjection": region.proxyInjection,
                        "packageRepositoryInjection": region.packageRepositoryInjection,
                        "certificateAuthorityInjection":
                            region.certificateAuthorityInjection,
```

**Review by garronej — CHANGES_REQUESTED** [review]
Hey @alexisdondon,  
Sorry for the long delay.  
This PR looks good to me!  

Just apply the suggested DIFF then I'll merge and release.

**Comment by garronej** [comment]
Je marge dès que c'est OK. Je reste attentif

**Comment by alexisdondon** [comment]
sorry i will check this today or tomorrow

File: src/core/secondaryAdapters/officialOnyxiaApiClient.ts (hunk 1 above)
Comment (alexisdondon) [reply src/core/secondaryAdapters/officialOnyxiaApiClient.ts]: ok when scripting this i asked myself if i should take this short path or the long way

**Comment by alexisdondon** [comment]
Ok i have accepted your suggestion commit

**Comment by alexisdondon** [comment]
I have rerun this in local to make another check it's ok for me

**Comment by garronej** [comment]
Thank you for your contribution.  
It's being released as I type theses words

**Pull request merged**